module-utils/utility/Utils.hpp

```diff
@@ -13,26 +13,106 @@
 #include <chrono>
 #include <random>
 #include <tuple>
+#include <type_traits>
 #include <functional>
 
 namespace utils
 {
     inline constexpr auto WHITESPACE = " \n\r\t\f\v";
-    constexpr unsigned int secondsInMinute =
-        std::chrono::duration_cast<std::chrono::seconds>(std::chrono::minutes(1)).count();
+    constexpr unsigned int secondsInMinute = 60;
 
-    std::string bytesToHex(const std::vector<std::uint8_t> &bytes);
-    std::vector<std::uint8_t> hexToBytes(const std::string &hex);
+    template <typename T> inline constexpr bool is_byte_v = std::is_integral_v<T> && sizeof(T) == sizeof(std::uint8_t);
 
-    template <typename T> std::string numToHex(T c)
+    template <typename T> using remove_cref_t = std::remove_const_t<std::remove_reference_t<T>>;
+
+    // NOTE: With short string optimization storing one byte hex in std::string is probably fine
+
+    template <typename T, std::enable_if_t<is_byte_v<T>, int> = 0> inline char halfByteToHex(T c)
+    {
+        return c < 10 ? '0' + c : 'a' + (c - 10);
+    }
+
+    // WARNING: Unsafe
+    template <typename T, std::enable_if_t<is_byte_v<T>, int> = 0>
+    inline void byteToHexAt(T c, std::string &result, std::size_t offset)
+    {
+        result[offset]     = halfByteToHex<std::uint8_t>((c & 0xF0) >> 4);
+        result[offset + 1] = halfByteToHex<std::uint8_t>(c & 0x0F);
+    }
+
+    template <typename T, std::enable_if_t<is_byte_v<T>, int> = 0> inline std::string byteToHex(T c)
```

Comment:
Some of these functions are not used anywhere. Do we need them or should I removed them?

Reply:
IMO those can be removed - if there will be need for them, then they will be created/reverted ;)

Reply:
I double-checked and it seems that:
- `byteToHex` is used in `QMBNManager.cpp`
- `bytesToHex` is not used
- `hexToByte` is not used
- `hexToBytes` is used in `QMBNManager.cpp` and `ServiceCellular.cpp` (added in this PR)

So only two of them would be removed.

Reply:
Done